### PR TITLE
Upgrade ESLint and @typescript-eslint to remove lint warning

### PR DESCRIPTION
**Pull request by jeremywiebe**
## Summary:

This PR bumps @typescript-eslint/* to v8.18.0 and ESLint to v8.57.1 to suppress a new lint warning after [bumping @khanacademy/eslint-plugi](https://github.com/Khan/perseus/pull/1949)n from v3.0.0 to v3.0.1. 

The warning looked like this: 

```
$ yarn lint 
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=4.7.4 <5.6.0

YOUR TYPESCRIPT VERSION: 5.7.2

Please only submit bug reports when using the officially supported version.

=============	
```

Issue: FEI-6012

## Test plan:

`$ yarn --check-files; yarn lint` no warnings about TypeScript versions

### CODE REVIEW

File: packages/math-input/src/components/input/math-input.tsx [hunk 1]
```diff
@@ -228,7 +227,7 @@ class MathInput extends React.Component<Props, State> {
                         // in which case we don't want to dismiss the keypad on check.
                         if (!isWithinKeypadBounds(x, y)) {
                             this.blur();
-                            this.props.onBlur && this.props.onBlur();
+                            this.props.onBlur?.();
```
Comment (kevinb-khan) [inline packages/math-input/src/components/input/math-input.tsx:230]: Nice!

File: packages/perseus-editor/src/widgets/interactive-graph-editor/locked-figures/util.ts [hunk 2]
```diff
@@ -151,6 +151,7 @@ export async function generateSpokenMathDetails(mathString: string) {
             case "specialCharacter":
                 // We don't want the backslash from special character
                 // to show up in the generated aria label.
+                // eslint-disable-next-line @typescript-eslint/no-unused-expressions
                 piece.content.length > 1
                     ? (convertedSpeech += piece.content.slice(1))
                     : (convertedSpeech += piece.content);
```
Comment (kevinb-khan) [inline packages/perseus-editor/src/widgets/interactive-graph-editor/locked-figures/util.ts:157]: You can configure `@typescript-eslint/no-unused-expressions` to allow ternaries if you want, see https://eslint.org/docs/latest/rules/no-unused-expressions#options.

File: packages/perseus/src/components/visibility-observer/visibility-observer.tsx [hunk 3]
```diff
@@ -52,7 +52,7 @@ function findPositionInScrollContainer(
                 : childRect.bottom < parentRect.top;
 
         return isAbove ? "hidden-above" : "hidden-below";
-    } catch (e) {
+    } catch {
         return "unknown";
     }
```
Comment (kevinb-khan) [inline packages/perseus/src/components/visibility-observer/visibility-observer.tsx:57]: I didn't realize this syntax was support.  Nice!

**Review by kevinb-khan — APPROVED** [review]
LGTM.  Nice use of optional chaining.  Thanks for doing this upgrade.

File: packages/perseus-editor/src/widgets/interactive-graph-editor/locked-figures/util.ts (hunk 2 above)
Comment (jeremywiebe) [reply packages/perseus-editor/src/widgets/interactive-graph-editor/locked-figures/util.ts:157]: I find assignment within a ternary to be pretty difficult to maintain/understand. I re-worked this slightly.

@nishasy for visibility as you wrote this. What do you think of transforming what you wrote to:

```typescript 
convertedSpeech +=
                    piece.content.length > 1
                        ? piece.content.slice(1)
                        : piece.content;
```

Comment (nishasy) [reply packages/perseus-editor/src/widgets/interactive-graph-editor/locked-figures/util.ts]: @jeremywiebe Good point! I like it 🚀

**Pull request merged**